README.md

```diff
@@ -11,3 +11,157 @@ Caching, aggregation, and relaying for xDS compliant clients and origin servers
 To get started:
 
 * [Contributing guide](CONTRIBUTING.md)
+
+## Design
+
+## Aggregation Rules
+
+xds-relay uses aggregation rules to determine how to aggregate inbound connections to the origin
+management server ([example](https://github.com/envoyproxy/xds-relay/blob/c83a956952c3e9762a08420ffc1c6bda5e0fff04/integration/testdata/keyer_configuration_e2e.yaml)
+).
+
+xDS requests that match under the same set of aggregation rules will fall into the same cache
+bucket. The cache is a mapping of a aggregation key to a xDS [DiscoveryResponse](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/discovery.proto#discoveryresponse)
+. This implies that all inbound xDS requests that map to the same aggregation key will retrieve the
+same DiscoveryResponse. This is useful in multi-cluster scenarios where xDS configuration does not
+differ greatly between clusters. For example, a service application scaled across 100 hosts will
+likely require the same data plane configurations across all hosts.
+
+At a high level, an aggregation rule is formed from `fragments`, `rules`, `match`, and `result`s.
```

Comment:
We should give a better description of how each one of these 4 components interact. For example, aggregations rules are composed of `fragments`, and a single `fragment` is composed of a list of `rules` where and each rule is composed of a pair composed of a `match` predicate and a `result` predicate.

Do you think it's too much to try to specify this using [EBNF](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form)? It might be, I'm just wondering if having that could alleviate the problem of being total in this document.